### ISSUE TITLE
Entity id already exists - ignoring: binary_sensor.golf_7_insured. Platform rdw does not generate unique IDs

### ISSUE THREAD

**Issue by golles** [eelcohn/home-assistant-rdw#21]
Hi,

Since a while I'm getting error messages like this:
```
Log Details (ERROR)
Logger: homeassistant.components.binary_sensor
Source: helpers/entity_platform.py:435
Integration: Binary sensor (documentation, issues)
First occurred: 12:09:02 PM (1 occurrences)
Last logged: 12:09:02 PM

Entity id already exists - ignoring: binary_sensor.golf_7_insured. Platform rdw does not generate unique IDs
```

On the configuration - integration page I had a lot of records for RDW, I deleted all beside the one with 3 entities and restarted (which game the above log) and the following UI:
![Screenshot 2020-04-30 at 12 12 20](https://user-images.githubusercontent.com/2211503/80699253-f3254b80-8adb-11ea-94a7-bd2f15429fb0.png)

Config:
```
❯ cat hass-config/packages/rdw.yaml 
rdw:
  - plate: 82ZXJ2
    name: "Golf 7"
    binary_sensors:
      - insured
    sensors:
      - expdate
      - recall

homeassistant:
  customize:
    sensor.golf_7_expdate:
      friendly_name: "APK Vervaldatum"
    binary_sensor.golf_7_insured:
      friendly_name: "Verzekeringsstatus"
    sensor.golf_7_recall:
      friendly_name: "Terugroepstatus"
  
group:
  AutoGolf7:
    name: Golf 7
    entities:
    - sensor.golf_7_expdate
    - binary_sensor.golf_7_insured
    - sensor.golf_7_recall
```

**Comment by eelcohn** [comment]
Can you tell me what version of Home Assistant you use, and what version of the RDW component you use?

[edit]
Try the following:
- make sure that you remove the rdw entry from your configuration.yaml
- remove all RDW component files from the custom_component directory
- remove ALL sensors, entities and other references
- reboot/restart Home Assistant (there shouldn't be any RDW references now since you deleted everything)
- Upgrade Home Assistant to the latest version
- Install the RDW component to the latest version (today on the 1st May 2020 is v2.9.3 the latest)
- Install the latest version of the RDW component in the custom_component directory
- Add the car(s) to Home Assistant using the UI / Add the cars to your configuration.yaml

**Comment by golles** [comment]
I followed your advice, on the first restart it looked all good, on every next restart another integration is added (like screenshot in 1st post).

It's still saying:
```
Entity id already exists - ignoring: binary_sensor.golf_7_insured. Platform rdw does not generate unique IDs
```
Am I missing something in my config?

PS. I've now changed the name field to `golf_7`, it was `"Golf 7"` (I didn't like the quotes there and result is the same).

**Comment by golles** [comment]
`❯ cat .storage/core.config_entries`

I see that `unique_id ` is set to `null`, (edit, but to be honest, I see many integrations have this).

```
...
            {
                "connection_class": "local_poll",
                "data": {
                    "binary_sensors": [
                        "insured"
                    ],
                    "dateformat": null,
                    "name": "golf_7",
                    "plate": "82ZXJ2",
                    "scan_interval": 3600,
                    "sensors": [
                        "expdate",
                        "recall"
                    ]
                },
                "domain": "rdw",
                "entry_id": "b9558bbf3d2c4102b0b4d632a2a11a92",
                "options": {},
                "source": "import",
                "system_options": {
                    "disable_new_entities": false
                },
                "title": "golf_7 (configuration.yaml)",
                "unique_id": null,
                "version": 1
            }
...
```

Another thing from the log:

```
Log Details (ERROR)
Logger: homeassistant.config_entries
Source: custom_components/rdw/__init__.py:152
First occurred: 3:20:25 PM (1 occurrences)
Last logged: 3:20:25 PM

Error unloading entry golf_7 (configuration.yaml) for rdw
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 295, in async_unload
    hass, self
  File "/config/custom_components/rdw/__init__.py", line 152, in async_unload_entry
    cancel = hass.data[DOMAIN][config_entry.data[CONF_PLATE]][DATA_LISTENER].pop(config_entry.entry_id)
KeyError: 'b9558bbf3d2c4102b0b4d632a2a11a92'
```

**Comment by eelcohn** [comment]
Thank you for your detailed explanation. I've been able to replicate the problem and I'm working on a solution. Not sure yet when it will be available, but it will definitely get fixed!

**Comment by golles** [comment]
Great to hear it helped. Now pulling your latest changes, happy to see you're fixing all the issues so quickly, bedankt!

**Comment by eelcohn** [comment]
Should be fixed in the latest release (v2.9.6).

**Comment by golles** [comment]
I've tested, indeed it's working now nicely. I like it that there is now 1 device with 3 entities, well done!
I had to delete all my existing RDW integrations (there were quite a few because of reboots) and now it works perfect, bedankt!
